### DISCUSSION TITLE
Segmentation fault (core dumped)

### DISCUSSION THREAD

**Discussion by Geo572** [neperfepx/neper#755] [Q&A] [answered]
Hello @rquey 
I have face problem like segmentation fault (core dumped)  if I put the value of mean 0.95 and 0.055 even my Neper current version is 4.6.0.
l am waiting for your kind suggestions.

Thank you.

neper -T -n 400 -morpho "diameq:lognormal(2,0.05),1-sphericity:lognormal(0.95,0.03)" -domain "square(22.5,45)" -dim 2 -format geo

========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.6.0
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 1 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2022, and GNU GPL'd, by Romain Quey.
Info   : No initialization file found (`/home/sajid/.neperrc').
Info   : ---------------------------------------------------------------
Info   : MODULE  -T loaded with arguments:
Info   : [ini file] (none)
Info   : [com line] -n 400 -morpho
         diameq:lognormal(2,0.05),1-sphericity:lognormal(0.95,0.03)
         -domain square(22.5,45) -dim 2 -format geo
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   : Creating domain...
Info   : Creating tessellation...
Info   :   - Setting seeds... 100%
Info   :   - Generating crystal orientations...
Info   :   - Running tessellation...
Info   :     > Initial solution: f   =475.029751260
Info   :     > Iter     1936762: fmin==170.897830789 f==170.934553085
Info   :     > Iter     1937964: fmin==170.897830789 f==171.460469569 (2 subplex)
Info   :     > Iter     2230244: fmin==169.956155724 f==169.956344878 (3 praxis)Segmentation fault (core dumped)

neper -T -n 400 -morpho "diameq:lognormal(2,0.05),1-sphericity:lognormal(0.055,0.03)" -domain "square(22.5,45)" -dim 2 -format geo

========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.6.0
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 1 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2022, and GNU GPL'd, by Romain Quey.
Info   : No initialization file found (`/home/sajid/.neperrc').
Info   : ---------------------------------------------------------------
Info   : MODULE  -T loaded with arguments:
Info   : [ini file] (none)
Info   : [com line] -n 400 -morpho
         diameq:lognormal(2,0.05),1-sphericity:lognormal(0.055,0.03)
         -domain square(22.5,45) -dim 2 -format geo
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   : Creating domain...
Info   : Creating tessellation...
Info   :   - Setting seeds... 100%
Info   :   - Generating crystal orientations...
Info   :   - Running tessellation...
Info   :     > Initial solution: f   =12.505178154
Info   :     > Iteration 605029: fmin=0.016769306 f=0.016769306Segmentation fault (core dumped)

**Comment by rquey** [comment, marked as answer]
In the first example, the sphericity is very low, which is probably not what you want. The second example runs fine with version 4.6.1:

```
$ neper -T -n 400 -morpho "diameq:lognormal(2,0.05),1-sphericity:lognormal(0.055,0.03)" -domain "square(22.5,45)" -dim 2 -format geo 

========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.6.1
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 17 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2022, and GNU GPL'd, by Romain Quey.
Info   : Loading initialization file `/home/rquey/.neperrc'...
Info   : ---------------------------------------------------------------
Info   : MODULE  -T loaded with arguments:
Info   : [ini file] -tesrformat ascii
Info   : [com line] -n 400 -morpho
         diameq:lognormal(2,0.05),1-sphericity:lognormal(0.055,0.03)
         -domain square(22.5,45) -dim 2 -format geo
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   : Creating domain...
Info   : Creating tessellation...
Info   :   - Setting seeds... 100%
Info   :   - Generating crystal orientations...
Info   :   - Running tessellation...
Info   :     > Initial solution: f   =12.505178154
Info   :     > Iteration 866721: fmin=0.015602191 f=0.015602191   
Info   :     > Final solution  : f   =0.015602191 (866721 iterations)
Info   :     > Reached `eps' criterion.
Info   : Writing results...
Info   :     [o] Writing file `n400-id1.geo'...
Info   :     [o] Wrote file `n400-id1.geo'.
Info   : Elapsed time: 652.096 secs.
========================================================================
```
